examples/identify.rs

```diff
@@ -0,0 +1,82 @@
+// Copyright 2018 Parity Technologies (UK) Ltd.
```

Comment:
Given that this example is focused on `libp2p-identify`, how about moving it to `protocols/identify/examples/`? To make sure one can still discover it, it can be mentioned in https://github.com/libp2p/rust-libp2p/blob/master/examples/README.md.

Reply:
I was not even aware that there are some examples placed together with some of the individual protocols. Is there a reason why not to have all the examples listed in `https://github.com/libp2p/rust-libp2p/tree/master/examples/*`? 

Having one canonical place where all the examples are listed, IMHO makes discovery easier.

Reply:
No strong opinion on my end. I see `/examples` as a folder for examples that combine many protocols and `protocols/xxx/examples` as a folder for examples targeting a single protocol.

> Having one canonical place where all the examples are listed, IMHO makes discovery easier.

I see your point.

Maybe @elenaf9 or @thomaseizinger have an opinion here.

Reply:
> I am leaning towards having examples focused on a single protocol in that protocol's folder. Since they are individual crates, imo they should "own" their own examples.

I share the same view. These protocols are individual crates and I think we should follow cargo conventions despite this being a large workspace.

> > I was not even aware that there are some examples placed together with some of the individual protocols.
> 
> We should definitely mention this in the top-level `examples/README.md`, which is currently not the case.

Another solution to this would be to do away with the current meta crate being "special" by being a workspace manifest and a crate at the same time. i.e. we could create a directory `meta` which could house the `libp2p` crate. This would also move the top-level `examples/` directory which might make it more obvious to users that they will find examples within each crate.